internal/policy/verify.go

```diff
@@ -530,105 +502,83 @@ func verifyEntry(ctx context.Context, repo *git.Repository, policy *State, attes
 	return nil
 }
 
-func verifyTagEntry(ctx context.Context, repo *git.Repository, policy *State, entry *rsl.ReferenceEntry) error {
-	// 1. Find authorized public keys for tag's RSL entry
-	trustedKeys, err := policy.FindPublicKeysForPath(ctx, fmt.Sprintf("git:%s", entry.RefName))
+func verifyTagEntry(ctx context.Context, repo *gitinterface.Repository, policy *State, attestationsState *attestations.Attestations, entry *rsl.ReferenceEntry) error {
+	// Find authorized public keys for tag's RSL entry
+	verifiers, err := policy.FindVerifiersForPath(fmt.Sprintf("%s:%s", gitReferenceRuleScheme, entry.RefName))
 	if err != nil {
 		return err
 	}
 
-	if len(trustedKeys) == 0 {
-		allKeys, err := policy.PublicKeys()
+	if len(verifiers) == 0 {
+		return nil
+	}
+
+	var authorizationAttestation *sslibdsse.Envelope
+	if attestationsState != nil {
+		authorizationAttestation, err = getAuthorizationAttestation(repo, attestationsState, entry)
```

Comment:
It looks like `attestationsState` is only used here. Can you add some more context on the difference between these two? My best guess without digging deeper is `attestationsState` is the set of all attestations for the point in time, and `authorizationAttestation` is the attestations that only applies to the current entry.

1. If the above description is correct, could there be multiple attestations for the entry that we might need to handle?
2. We should move this nil-check to getAuthorizationAttestation if it's something that we expect to be done each time.